tailnet/resume_test.go

```diff
@@ -121,61 +24,76 @@ func TestResumeTokenKeyProvider(t *testing.T) {
 		t.Parallel()
 
+		ctx := testutil.Context(t, testutil.WaitShort)
 		id := uuid.New()
 		clock := quartz.NewMock(t)
-		provider := tailnet.NewResumeTokenKeyProvider(key, clock, tailnet.DefaultResumeTokenExpiry)
-		token, err := provider.GenerateResumeToken(id)
+		provider := tailnet.NewResumeTokenKeyProvider(newKeySigner(key), clock, tailnet.DefaultResumeTokenExpiry)
+		token, err := provider.GenerateResumeToken(ctx, id)
 		require.NoError(t, err)
 		require.NotNil(t, token)
 		require.NotEmpty(t, token.Token)
 		require.Equal(t, tailnet.DefaultResumeTokenExpiry/2, token.RefreshIn.AsDuration())
 		require.WithinDuration(t, clock.Now().Add(tailnet.DefaultResumeTokenExpiry), token.ExpiresAt.AsTime(), time.Second)
 
-		// Advance time past expiry
-		_ = clock.Advance(tailnet.DefaultResumeTokenExpiry + time.Second)
+		// Advance time past expiry. Account for leeway.
+		_ = clock.Advance(tailnet.DefaultResumeTokenExpiry + time.Second*61)
 
-		_, err = provider.VerifyResumeToken(token.Token)
-		require.ErrorContains(t, err, "expired")
+		_, err = provider.VerifyResumeToken(ctx, token.Token)
+		require.Error(t, err)
+		require.ErrorIs(t, err, jwt.ErrExpired)
 	})
 
 	t.Run("InvalidToken", func(t *testing.T) {
 		t.Parallel()
 
-		provider := tailnet.NewResumeTokenKeyProvider(key, quartz.NewMock(t), tailnet.DefaultResumeTokenExpiry)
-		_, err := provider.VerifyResumeToken("invalid")
+		ctx := testutil.Context(t, testutil.WaitShort)
+		provider := tailnet.NewResumeTokenKeyProvider(newKeySigner(key), quartz.NewMock(t), tailnet.DefaultResumeTokenExpiry)
+		_, err := provider.VerifyResumeToken(ctx, "invalid")
 		require.ErrorContains(t, err, "parse JWS")
 	})
 
 	t.Run("VerifyError", func(t *testing.T) {
 		t.Parallel()
 
+		ctx := testutil.Context(t, testutil.WaitShort)
 		// Generate a resume token with a different key
 		otherKey, err := tailnet.GenerateResumeTokenSigningKey()
 		require.NoError(t, err)
-		otherProvider := tailnet.NewResumeTokenKeyProvider(otherKey, quartz.NewMock(t), tailnet.DefaultResumeTokenExpiry)
-		token, err := otherProvider.GenerateResumeToken(uuid.New())
+		otherSigner := newKeySigner(otherKey)
+		otherProvider := tailnet.NewResumeTokenKeyProvider(otherSigner, quartz.NewMock(t), tailnet.DefaultResumeTokenExpiry)
+		token, err := otherProvider.GenerateResumeToken(ctx, uuid.New())
 		require.NoError(t, err)
 
-		provider := tailnet.NewResumeTokenKeyProvider(key, quartz.NewMock(t), tailnet.DefaultResumeTokenExpiry)
-		_, err = provider.VerifyResumeToken(token.Token)
-		require.ErrorContains(t, err, "verify JWS")
+		signer := newKeySigner(key)
+		signer.ID = otherSigner.ID
+		provider := tailnet.NewResumeTokenKeyProvider(signer, quartz.NewMock(t), tailnet.DefaultResumeTokenExpiry)
+		_, err = provider.VerifyResumeToken(ctx, token.Token)
+		require.ErrorIs(t, err, jose.ErrCryptoFailure)
 	})
 }
+
+func newKeySigner(key tailnet.ResumeTokenSigningKey) cryptokeys.StaticKey {
+	return cryptokeys.StaticKey{
+		ID:  uuid.New().String(),
```

Comment:
I know these are just strings, but for consistency, we should use integers not UUID to match what we actually use in production.